### DISCUSSION TITLE
Loader V7  about

### DISCUSSION THREAD

**Discussion by gonulbuya** [pixijs/pixijs#9433] [General]
Why is the loader in pixijs version 7 different from version 6?

const loader = PIXI.Loader.shared; // PixiJS exposes a premade instance for you to use.
// or
const loader = new PIXI.Loader(); // You can also create your own if you want

i couldn't run them in pixijs 7

**Comment by bigtimebuddy** [comment]
Loader was removed in v7 favor of new Assets API (uses promises, simpler caching, support for asset bundles)

**Reply by gonulbuya** [reply]
We should use the following code to upload multiple images. Is there another way? Or should we just use the code below?

const app = new PIXI.Application({ background: '#1099bb' });
document.body.appendChild(app.view);

// Add the assets to load
PIXI.Assets.add('flowerTop', 'examples/assets/flowerTop.png');
PIXI.Assets.add('eggHead', 'examples/assets/eggHead.png');

// Load the assets and get a resolved promise once both are loaded
const texturesPromise = PIXI.Assets.load(['flowerTop', 'eggHead']); // => Promise<{flowerTop: Texture, eggHead: Texture}>

// When the promise resolves, we have the texture!
texturesPromise.then((textures) => {
     // create a new Sprite from the resolved loaded Textures

     const flower = PIXI.Sprite.from(textures.flowerTop);
     flower.anchor.set(0.5);
     flower.x = app.screen.width * 0.25;
     flower.y = app.screen.height / 2;
     app.stage.addChild(flower);

     const egg = PIXI.Sprite.from(textures.eggHead);
     egg.anchor.set(0.5);
     egg.x = app.screen.width * 0.75;
     egg.y = app.screen.height / 2;
     app.stage.addChild(egg);
});

**Reply by bigtimebuddy** [reply]
Yup that's it

**Reply by gonulbuya** [reply]
Thank you very much.
